### PR TITLE
fix(#patch); abra; fix holey array

### PR DESCRIPTION
**Pull request by dmelotik**
Fixing holey array issue in abra: #1617

- Build using different structure
- Testing: https://okgraph.xyz/?q=dmelotik%2Fabracadabra-ethereum

### CODE REVIEW

File: subgraphs/abracadabra/src/common/tokens.ts [hunk 1]
```diff
@@ -76,108 +76,103 @@ export function fetchTokenDecimals(tokenAddress: Address): i32 {
...
-  } else {
-    return INVALID_TOKEN_DECIMALS as i32;
   }
+
+  return INVALID_TOKEN_DECIMALS as i32;
 }
 
 export function isNullEthValue(value: string): boolean {
-  return value == "0x0000000000000000000000000000000000000000000000000000000000000001";
+  if (value == "0x0000000000000000000000000000000000000001" || value == "0x0000000000000000000000000000000000000000")
```
Comment (jaimehgb) [inline subgraphs/abracadabra/src/common/tokens.ts:88]: Why `….0001` is considered null?

**Review by jaimehgb — COMMENTED** [review]
What’s exactly the fix? 😅

File: subgraphs/abracadabra/src/common/tokens.ts (hunk 1 above)
Comment (dmelotik) [reply subgraphs/abracadabra/src/common/tokens.ts:88]: It's a burn address: https://etherscan.io/address/0x0000000000000000000000000000000000000001

**Comment by dmelotik** [comment]
@jaimehgb 

> What’s exactly the fix?
redoing the static class, it seems to be causing a holey array issue, and I added an extra check for burn addresses

**Review by jaimehgb — APPROVED** [review]

**Comment by dmelotik** [comment]
@jaimehgb we should probably get down to the bottom of this. I have never known what is actually wrong when I get the holey array error. Are you aware?

**Comment by dmelotik** [comment]
Aha, thanks @jaimehgb

**Pull request merged**